cucumber-expressions/java/src/main/java/io/cucumber/cucumberexpressions/TreeRegexp.java

```diff
@@ -24,13 +24,11 @@
 
     TreeRegexp(Pattern pattern) {
         this.pattern = pattern;
-        this.groupBuilder = createGroupBuilder(pattern);
+		this.groupBuilder = createGroupBuilder(pattern.pattern());
     }
 
-    private static GroupBuilder createGroupBuilder(Pattern pattern) {
```

Comment:
This method assumes that a valid regex is provided. Rather then dealing with parsing a regex internally we can ensure the regex is valid by making the caller deal with that.

Reply:
Its just a bit strange to require parsing it into a regexp and then unwraping it to get the text :-)
I would argue that passing invalid data on the parser does not necessarily yields valid results.

Reply:
A method should have well defined behavior. In this case its behavior is not well defined when the precondition of a valid regex is violated. We can enforce this by using the type system. That this results in parsing a regex, and then parsing the regex again doesn't seem relevant.